### DISCUSSION TITLE
Indexing pop up window

### DISCUSSION THREAD

**Discussion by miquecg** [neoclide/coc.nvim#3920] [Q&A]
Hi,

I've started using ErlangLS with coc.nvim and so far I've noticed that every time I open an Erlang file or type in the buffer, several pop up windows quickly appear and disappear. I think they report the indexing progress of the project, usually rather quick.

So the question is, how can I disable this behaviour? Is it some setting on the server? Because I don't experience similar behaviour with other Language Servers configured, mainly ElixirLS.

This is my config:

```json
  "languageserver": {
    "erlang": {
      "command": "erlang_ls",
      "filetypes": ["erlang"]
    }
```

**Comment by fannheyward** [comment]
Add `"notification.disabledProgressSources": ["erlang"]` in your coc-settings.json.

**Reply by miquecg** [reply]
Doesn't help :(

**Reply by fannheyward** [reply]
I haven't use Erlang, what's the output of `:CocList services`  when you use `Erlang_ls`?

**Reply by miquecg** [reply]
Interesting, I see two services:

```
  languageserver.gleam [init] gleam
* languageserver.erlang [running] erlang
```

This is the relevant config:

```json
  "languageserver": {
    "erlang": {
      "command": "erlang_ls",
      "filetypes": ["erlang"]
    },
    "gleam": {
      "command": "gleam",
      "args": ["lsp"],
      "filetypes": ["gleam"],
      "rootPatterns": ["gleam.toml", ".git"]
    }
  }
```

Gleam is another language on the BEAM that compiles to Erlang.

**Reply by miquecg** [reply]
@fannheyward After further analysis it looks like coc.nvim has workdone-progress-reports enabled.

```
        "window": {
            "showMessage": {
                "messageActionItem": {
                    "additionalPropertiesSupport": true
                }
            },
            "showDocument": {
                "support": true
            },
            "workDoneProgress": true
        },
```

Is it possible to disable these reports globally? As I've said, `"notification.disabledProgressSources": ["erlang"]` doesn't make any difference.

**Reply by miquecg** [reply]
Per your [own response](https://github.com/neoclide/coc.nvim/issues/3875#issuecomment-1147312356) I see that a few months ago it was not possible to configure this behaviour.

**Reply by chemzqm** [reply]
The source name need prefix with language., you can try `[*]`

**Reply by miquecg** [reply]
Thanks, `"notification.disabledProgressSources": ["*"]` worked and pop-ups are gone now.

> The source name need prefix with language.

I don't quite understand what you mean. Can you give an example please? Why setting this config to `["erlang"]` was not having the same result?

**Reply by chemzqm** [reply]
Because it's id of configured language server, use `:call coc#notify#show_sources()` to show source names.

**Reply by miquecg** [reply]
> `:call coc#notify#show_sources()`

I don't get any output from this.

**Reply by chemzqm** [reply]
Your notification windows updated with source name after you call the function

**Reply by miquecg** [reply]
Sorry but I don't see any notification window. Maybe I have to disable the setting before doing this? I'm on vim

**Reply by miquecg** [reply]
I think I've found it somewhere else:

```
2022-09-20T16:08:55.599 INFO (pid:212706) [language-client-index] - Language server "languageserver.erlang" started with 212942
```

**Reply by miquecg** [reply]
None of these configs seem to work though:

> `"notification.disabledProgressSources": ["languageserver.erlang"]`
> `"notification.disabledProgressSources": ["languageserver.erlang.*"]`

**Reply by chemzqm** [reply]
We will use status line by default, so it's not needed by default.

**Reply by miquecg** [reply]
Sorry, I've edited my last comment. Just to be clear, ¿is it possible to disable progress just for ErlangLS? To me it looks like not at the moment. I think the only way to make this work is to stick with `["*"]`.
